### ISSUE TITLE
Iterating over array_keys([]) is analyzed differently from iterating over []

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#384]
`empty` isn't a type that `$key` can have, I'd expect mixed or something else.

The original example had psalm failing to pick up possible types that were added to the array in question.

```php
<?php

function expect_string(string $x) : void {
    echo $x;
}

function test(array $inputs) : void {
    foreach (array_keys([]) as $key) {
        expect_string($key);  // Unexpected InvalidScaarArgument: Expects string, empty provided
    }
    foreach ([] as $key2 => $v2) {
        // No issue is emitted for these two lines
        expect_string($key2);
        expect_string($v2);
    }
}
```

**Comment by muglug** [comment]
Fixed in https://github.com/vimeo/psalm/commit/31ce7166478f46c8f5a69b22cf7c1d3306869fe2
